### PR TITLE
adds diff_drive simulation part to report/sections and the simulation profile for the diff_derive under simulations

### PR DESCRIPTION
**Pull request by WastedWaffle**
this merge will add:
mkdir simulations: the matlab and simulink files for diff_drive
pictures to report/sections/assets
diff-drive simulation in to report/sections/system-design.tex

### CODE REVIEW

File: report/sections/system-design.tex [hunk 1]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+\label{fig:Theta_Step}
+\end{figure}
+
+The result of the simulation, the step response for the angular position control system is presented in figure(\ref{fig:Theta_Step}).
+As it shown this is a stable system, it has no steady state error, no oscillation and relative fast rise-time.\\
+
+In this part of the mobile platforms movement it is important to be able to control the spacial position $(x, y)$. 
+As shown in the kinematic equations Eq.(\ref{eq:MotionEq}) the positions are dependent on both the angular velocity $\omega$ and the average velocity $v$. 
+A design of a controller for the average velocity is required to make it possible to control the spacial position. 
+Take note of the two input $AverageVelocity$, $Desired\_Theta$ and the two output $X$, $Y$ in figure(\ref{fig:PID1}).\\
```
Comment (tmplt) [inline report/sections/system-design.tex]: `$ ... $` is used for math mode. When referring to variables used in code, used `\texttt{...}` instead.

File: report/sections/system-design.tex [hunk 2]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+\label{eq:cart2polar}
+\end{eqnarray}
+\\
+
+Now by defining the distance as an error to a PID controller for the average velocity, it makes it possible for the mobile platform to stop at the desired position. 
+In this case a P controller, a controller which has only a proportional gain, is enough.\\
+\begin{figure*}[ht]
+\centering
+\includegraphics[width=\textwidth]{sections/assets/V_PID.PNG}
+\caption{Overview of the complete systems block diagram for the spatial position control. figure(\ref{fig:PID1}) is an overview of the "Steering" block and Eq.(\ref{eq:cart2polar}) is the mathematical definition for the block function "Cartesian- to Polar-coordinates Converter".}
```
Comment (tmplt) [inline report/sections/system-design.tex]: Latex doesn't render `"some quotation` very nicely, use ` ``some quotation'' ` instead.

File: report/sections/system-design.tex [hunk 3]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
 \subsection{Simulation}
 % Simulte the models from the previous section and show that it will work.
 % Motivate regulation approach.
+\subsubsection{Mobile platform}
+A simulation profile is made to make sure of the unicycle mode in sub section \ref{subsubsec:unicycle} is the suitable model for this application.
+The simulation is made in discrete time domain with the sampling frequency of $10Hz$.
+This is done in the described way due to the robot is going to operate in discrete time.
+The sampling frequency is assumed to be low and that is because the designed controllers are going to be able to work fine for higher samplings frequencies but that is not guaranteed for vise-versa.\\  
+As the unicycle model describes the systems dynamics.
+The position of the robot $(x, y, \theta)$ is dependent on the two wheel velocities. Which in turn has dependencies to the average velocity $V$ and the angular velocity $\omega$. These relations are presented in Eq.(\ref{eq:ThetaDot}) - Eq.(\ref{eq:av_velocity}).\\
```
Comment (tmplt) [inline report/sections/system-design.tex]: * `\eqref`are commonly used: they wrap the number in parantheses.
* Use `--` to denote ranges, so change to `Eq.~\eqref{eq:ThetaDot}--Eq.~\eqref{eq:av_velocity}`.

File: report/sections/system-design.tex [hunk 4]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+v = \frac{v_R + v_L}{2} 
+\label{eq:av_velocity}
+\end{equation}
 
 \subsection{Simulation}
 % Simulte the models from the previous section and show that it will work.
 % Motivate regulation approach.
+\subsubsection{Mobile platform}
+A simulation profile is made to make sure of the unicycle mode in sub section \ref{subsubsec:unicycle} is the suitable model for this application.
+The simulation is made in discrete time domain with the sampling frequency of $10Hz$.
```
Comment (tmplt) [inline report/sections/system-design.tex]: Units are not to be in italic; rewrite `$10Hz$` as `$10$~Hz`.

File: report/sections/system-design.tex [hunk 5]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+The position of the robot $(x, y, \theta)$ is dependent on the two wheel velocities. Which in turn has dependencies to the average velocity $V$ and the angular velocity $\omega$. These relations are presented in Eq.(\ref{eq:ThetaDot}) - Eq.(\ref{eq:av_velocity}).\\
+\begin{figure*}[ht]
+\centering
+\includegraphics[width=\textwidth]{sections/assets/omegaCtrlr.png}
+\caption{Overview of the systems block diagram where the angular velocity $\omega$ is controlled by a PID controller.}
+\label{fig:overview}
+\end{figure*} 
+
+A pole placement analyze is required for the design of a PID controller.
+This analyze is done in continues time domain, due to its simplicity.\\
```
Comment (tmplt) [inline report/sections/system-design.tex]: Don't specity `\\` (linebreak) here or in other paragraphs. Paragraph typesetting is better left to LaTeX itself. If it looks bad/weird without them we'll do a once-over when finalizing the report and handle any manual line breaks if needed then.

File: report/sections/system-design.tex [hunk 6]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+
+By taking the Laplace transform of Eq.(\ref{eq:omega}) gives:
+\begin{equation}
+s\cdot \Theta(s)= \Omega(s)
+\end{equation}
+
+Where $G(s)$ is the transfer function
+\begin{equation}
+\therefore G(s)=\frac{\Omega(s)}{s\cdot\Theta(s)} = 1
+\end{equation}
```
Comment (tmplt) [inline report/sections/system-design.tex:110]: Everything from line 101 down to here seems to be in a single sentence. Remove the empty lines to that all flowing text is considered a single paragraph.

File: report/sections/system-design.tex [hunk 7]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+        \draw [draw,->] (input) -- node {$U(s)$} (sum);
+        \draw [->] (sum) -- node {} (controller);
+        \draw [->] (controller) -- node {} (plant);
+        \draw [->] (plant) -- node [name=y] {$Y(s)$}(output);
+        \draw [->] (y) |- node [above,pos=0.79] {} (feedback) ;
+        \draw [->] (feedback) -| node[pos=0.99] {$-$} 
+        node [near end] {} (sum);
+        \end{tikzpicture}
+    \end{center}
+    \caption{Closed Loop system with a integrator in the feedback.}\label{fig:closedloop}
```
Comment (tmplt) [inline report/sections/system-design.tex]: `Closed Loop` -> `Closed loop`.

File: report/sections/system-design.tex [hunk 8]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+By using a function called arctan2 or atan2 in many programming languages it is possible to accomplish this challenge.\\
+This function often has two input argument (y,x).\\ 
+By defining 
+\begin{equation}
+\theta = atan2(sin(\theta),\, cos(\theta))
+\end{equation}
+$\theta$ will always be in $(-\pi,\pi]$ span.\\
+
+Due to the physical limits the servos have, two saturation blocks are required.
+These blocks are shown in figure(\ref{fig:PID1}) and the assumption for their limits are (-10,10) $[cm/sec]$.
```
Comment (tmplt) [inline report/sections/system-design.tex]: Units are not to be italicized. Use SI units. `(-10,10) $[cm/sec]$` -> `$(-10,10)$~[cm/s]`.

File: report/sections/system-design.tex [hunk 9]
```diff
@@ -34,10 +35,205 @@ \subsection{Model}
...
+Now by defining the distance as an error to a PID controller for the average velocity, it makes it possible for the mobile platform to stop at the desired position. 
+In this case a P controller, a controller which has only a proportional gain, is enough.\\
+\begin{figure*}[ht]
+\centering
+\includegraphics[width=\textwidth]{sections/assets/V_PID.PNG}
+\caption{Overview of the complete systems block diagram for the spatial position control. figure(\ref{fig:PID1}) is an overview of the "Steering" block and Eq.(\ref{eq:cart2polar}) is the mathematical definition for the block function "Cartesian- to Polar-coordinates Converter".}
+\label{fig:V_PID}
+\end{figure*}
+ 
+Here it comes some of the results for the spacial position control simulation. In this case the desired position is (100, 100), the proportional gain for the angular velocity controller is $K_p = 4$ and the proportional gain for the average velocity controller is $K_p = 5$. If you are interested to see more plots, you can go to the Git-repository under simulation/Diff\_Drive/read\_me and you will find out how to generate more relative plots for this system.\\
```
Comment (tmplt) [inline report/sections/system-design.tex]: * "Here it comes some" -> "Here comes some".
* `(100, 100)` -> `$(100, 100)$` if we want to be consisent with how we typeset elements (we do).
* The repository path should be specified in `\texttt{...}`.

File: simulations/Diff_Drive/read_me.txt [hunk 10]
```diff
@@ -0,0 +1,3 @@
+step 1: to initialize the simulink profile you need to run the first block in plotScript.mlx
+step 2: run the simulink profile diff_drive.slx
+step 3: run the second part of the plotScript.mlx to get the rest of the plots
```
Comment (tmplt) [inline simulations/Diff_Drive/read_me.txt]: Rename this file to `README.txt`. With that name GitHub automatically renders the file if we enter `simulations/`.

**Review by tmplt — CHANGES_REQUESTED** [review]
* `assets/Theta_StepResponse`: syatem -> system;
* ditto in `assets/Theta_pi_stepresponce` (also responce -> response);
* `assets/WV`: velovities -> velocities;

File: report/sections/system-design.tex (hunk 5 above)
Comment (WastedWaffle) [reply report/sections/system-design.tex]: should I remove all \\ (linebreak) in the section?

Comment (tmplt) [reply report/sections/system-design.tex]: Remove all `\\` in the file.

**Review by tmplt — DISMISSED** [review]

**Review by tmplt — APPROVED** [review]

**Pull request merged**